usr/libexec/greenboot/greenboot

```diff
@@ -82,10 +64,12 @@ case "$1" in
   "check")
     rc=0
     for health_check_path in "${SCRIPTS_CHECK_PATHS[@]}"; do
-      script_runner "$health_check_path/required.d" "strict" "Running Required Health Check Scripts..." || rc=1
+      script_runner "$health_check_path/required.d" "strict" "Running Required Health Check Scripts..." || {
```

Comment:
I understand the logic part here but why do we need OR (||) operator here?  
can we not just use something like
If return val of script_runner !=0 i.e failure then break .
(or is it the bash thing !?)

Reply:
both approaches yields same result, I don't see what `if..then ..` offers more?